client/src/connection_cache.rs

```diff
@@ -167,7 +167,7 @@ struct ConnectionMap {
 impl ConnectionMap {
     pub fn new() -> Self {
         Self {
-            map: LruCache::new(MAX_CONNECTIONS),
+            map: HashMap::with_capacity(MAX_CONNECTIONS),
```

Comment:
How do we bound the HashMap size? I do not see we remove from the map.

Reply:
Yes, figuring out the strategy to evict entries when we hit the max limit. See comments from @ryleung-solana